src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getAverageGrade(grades: number[]): number {
+  return grades.reduce((a, b) => a + b, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+
+  if (sortBy === SortType.Name || sortBy === SortType.Surname) {
+    return order === 'asc'
+      ? studentsCopy.sort(
+        (current, next) => current[sortBy].localeCompare(next[sortBy]),
+      )
+      : studentsCopy.sort(
+        (current, next) => next[sortBy].localeCompare(current[sortBy]),
+      );
```

Comment:
You can always return sorted array and change sorting inside callback when you have different conditions
```suggestion
    return studentsCopy.sort((current, next) => (
      sortType === 'asc'
        ? current[sortBy].localeCompare(next[sortBy])
        : next[sortBy].localeCompare(current[sortBy])
    ));
```